packages/accounts/src/msca/plugins/multisig/signer.ts

```diff
@@ -0,0 +1,134 @@
+  type TypedData,
+  type TypedDataDefinition,
+  encodeAbiParameters,
+  hexToBigInt,
+} from "viem";
+import { MultisigPlugin, MultisigPluginAbi } from "./plugin.js";
+import type { Signature } from "./types.js";
+import { UserOpSignatureType, SignerType } from "./types.js";
+
+export const multisigMessageSigner = <
+  TTransport extends Transport,
+  TSigner extends SmartAccountSigner
+>(
+  client: BundlerClient<TTransport>,
+  accountAddress: Address,
+  signer: () => TSigner,
+  pluginAddress: Address = MultisigPlugin.meta.addresses[client.chain.id]
+) => {
+  const signWith712Wrapper = async (msg: Hash): Promise<`0x${string}`> => {
+    const [, name, version, chainId, verifyingContract, salt] =
+      await client.readContract({
+        abi: MultisigPluginAbi,
+        address: pluginAddress,
+        functionName: "eip712Domain",
+        account: accountAddress,
+      });
+
+    return signer().signTypedData({
+      domain: {
+        chainId: Number(chainId),
+        name,
+        salt,
+        verifyingContract,
+        version,
+      },
+      types: {
+        AlchemyModularAccountMessage: [{ name: "message", type: "bytes" }],
+      },
+      message: {
+        message: msg,
+      },
+      primaryType: "AlchemyModularAccountMessage",
+    });
+  };
+
+  return {
+    getDummySignature: async (): Promise<`0x${string}`> => {
+      const [, threshold] = await client.readContract({
+        abi: MultisigPluginAbi,
+        address: pluginAddress,
+        functionName: "ownershipInfoOf",
+        args: [accountAddress],
+      });
+
+      // (uint upperLimitPreVerificationGas, uint upperLimitMaxFeePerGas, uint upperLimitMaxPriorityFeePerGas)
+      // all sigs will be on "actual" with v = 32
+      return (
+        "0x" +
+        "FF".repeat(64 * 3) +
+        "FF".repeat(Number(threshold) * 39) +
+        "20"
+      ).repeat(Number(threshold)) as `0x${string}`;
+    },
+
+    signUserOperationHash: (uoHash: `0x${string}`): Promise<`0x${string}`> => {
+      return signer().signMessage({ raw: uoHash });
+    },
+
+    signMessage({
+      message,
+    }: {
+      message: SignableMessage;
+    }): Promise<`0x${string}`> {
+      return signWith712Wrapper(hashMessage(message));
+    },
+
+    signTypedData: <
+      const typedData extends TypedData | Record<string, unknown>,
+      primaryType extends keyof typedData | "EIP712Domain" = keyof typedData
+    >(
+      typedDataDefinition: TypedDataDefinition<typedData, primaryType>
+    ): Promise<Hex> => {
+      return signWith712Wrapper(hashTypedData(typedDataDefinition));
+    },
+  };
+};
+
+export const formatSignatures = (signatures: Signature[]) => {
+  let eoaSigs: string = "";
+  let contractSigs: string = "";
+  let offset: bigint = BigInt(65 * signatures.length);
+  signatures
+    .sort((a, b) => {
+      const bigintA = hexToBigInt(a.signer);
+      const bigintB = hexToBigInt(b.signer);
+
+      return bigintA < bigintB ? -1 : bigintA > bigintB ? 1 : 0;
```

Comment:
nit: this is effectively `compareTo` in Java. Might be good to make this a util method? that way, we can also simplify this nested ternary - it's a bit hard to parse.

so funny there isn't a method in native TS that does this squashing like lower level languages lol

Reply:
IIRC the issue with native comparisons is that address is checksummed in an Eth format and gives u random capitalizations, so string comparison won't work out of the box. But addr.toLower() < addr.toLower() probably does

Good idea with utils, this was what we currently do in multiOwner

Reply:
Is there a concern with runtime speed here? Not fully sure with js engine but this might be slightly inefficient with the multiple string concats vs preallocating the full length and copying bytes in manually. Another way of doing this is arr.map() returning 2 strings per element, then combining it with arr.reduce. Curious to hear which one you'd do